testing.rb

```diff
@@ -0,0 +1,311 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockNesting,Metrics/MethodLength,Metrics/ModuleLength,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity,Metrics/AbcSize
+
+# module Enumerable
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = to_a if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n])
+      n += 1
+    end
+    arr
+  end
```

Comment:
Your #my_each method should return the range itself after calling the given block when a block is given and self is a range.  
If you notice, line 10 - 12 converts both hash or range to an array,  but if a range/hash is given as a block it should return a range/hash as well.... changing line 18 to `self` should fix this.
This applies to #my_each_with_index as well.